src/main/java/org/jboss/staxmapper/XMLMapperImpl.java

```diff
@@ -100,10 +108,24 @@ public void deparseDocument(XMLContentWriter contentWriter, XMLStreamWriter stre
         contentWriter.writeContent(new FormattingXMLStreamWriter(streamWriter));
     }
 
-    @SuppressWarnings({ "unchecked" })
+    private XMLElementReader<?> getParser(final QName name) {
+        rootElements.computeIfAbsent(name, qName -> {
+            XMLElementReader<?> reader = rootElementsProvider.get(name).get(); //lazy create of parser
+            return new WeakReference<>(reader);
```

Comment:
What is the point of this weak ref?  Won't the provider reference any classes that its reader references?